### PR TITLE
Fix WFs filtering and add a cache for a better performance

### PR DESCRIPTION
**Pull request by marounkoussaifi**

### CODE REVIEW

File: src/integration-test/java/org/ow2/proactive/catalog/rest/controller/BucketControllerIntegrationTest.java [hunk 1]
```diff
@@ -309,15 +309,15 @@ public void testListBucketsGivenPrefixKindAndEmptyBucket() throws UnsupportedEnc
                .then()
                .assertThat()
                .statusCode(HttpStatus.SC_OK)
-               .body("", hasSize(2));
+               .body("", hasSize(1));
```
Comment (fviale) [inline src/integration-test/java/org/ow2/proactive/catalog/rest/controller/BucketControllerIntegrationTest.java:312]: you need to update the above comments

File: src/integration-test/java/org/ow2/proactive/catalog/service/BucketServiceIntegrationTest.java [hunk 2]
```diff
@@ -264,12 +264,12 @@ public void testListBucketsCaseInsensitiveFilterByKindPrefix() {
         List<BucketMetadata> bucketMetadatasWorkflows = bucketService.listBuckets(null,
                                                                                   Optional.of("WORKFLOW"),
                                                                                   Optional.empty());
-        assertThat(bucketMetadatasWorkflows).hasSize(4);
+        assertThat(bucketMetadatasWorkflows).hasSize(3);
```
Comment (fviale) [inline src/integration-test/java/org/ow2/proactive/catalog/service/BucketServiceIntegrationTest.java:267]: you need to update the above comment

File: src/integration-test/java/org/ow2/proactive/catalog/service/BucketServiceIntegrationTest.java [hunk 3]
```diff
@@ -334,13 +334,13 @@ public void testListBucketsFilterByKindList() {
         List<BucketMetadata> bucketMetadatasWorkflow = bucketService.listBuckets(null,
                                                                                  Optional.of("Workflow,Workflow/PCA"),
                                                                                  Optional.empty());
-        assertThat(bucketMetadatasWorkflow).hasSize(3);
+        assertThat(bucketMetadatasWorkflow).hasSize(2);
```
Comment (fviale) [inline src/integration-test/java/org/ow2/proactive/catalog/service/BucketServiceIntegrationTest.java:337]: same remark

**Review by fviale — APPROVED** [review]
please update the test comments, otherwise LGTM

**Review by mboussaa — APPROVED** [review]

**Pull request merged**